src/pyodide/internal/snapshot.ts

```diff
@@ -264,8 +264,13 @@ function memorySnapshotDoImports(Module: Module): string[] {
         MetadataReader.getWorkerFiles('py')
       )
     );
-
-  const deduplicatedModules = [...new Set(importedModules)];
+  const moduleSet = new Set(importedModules);
+  // Importing pyodide before `finalizeBootstrap()` is called causes mayhem
+  // Make sure we don't do that!
+  // TODO: in Pyodide 0.27 we can call finalizeBootstrap() before taking the snapshot. Doing so will
```

Comment:
Since this is only necessary for 0.26, should we gate this behind a version check?

Reply:
It's necessary for both until we move `finalizeBootstrap()`.